components/script/dom/idbrequest.rs

```diff
@@ -116,7 +131,16 @@ impl RequestListener {
                 IdbResult::Key(key) => {
                     key_type_to_jsval(GlobalScope::get_cx(), &key, answer.handle_mut())
                 },
-                IdbResult::Data(serialized_data) => {
+                IdbResult::Keys(keys) => {
+                    let mut array: Vec<JSVal> = Vec::with_capacity(keys.len());
```

Comment:
We need to use rooted_vec here, like https://github.com/servo/servo/blob/ae5b40ebf9cc02c473836cadf733b041cf4d0ec7/components/script/indexed_db.rs#L60 . The key_type_to_jsval call could cause the prior members of the array to be invalidated if a GC occurs.